src/environmentd/src/http/static/js/memory.jsx

```diff
@@ -27,52 +27,148 @@ async function query(sql) {
+        setSqlResponse(results);
+        if (!replicaName && results.length > 0) {
+          setCurrentClusterName(results[0][0]);
+          setCurrentReplicaName(results[0][1]);
+        }
         setLoading(false);
       })
       .catch((error) => {
         setError(error);
         setLoading(false);
       });
-  }, [sql]);
+  }, []);
 
-  return [response, loading, error];
+  useEffect(() => {
+    if (!currentReplicaName) return;
+    const params = new URLSearchParams(location.search);
+    params.set('cluster_name', currentClusterName);
+    params.set('replica_name', currentReplicaName);
+    window.history.replaceState({}, '', `${location.pathname}?${params}`);
+  }, [currentClusterName, currentReplicaName])
+
+  return (
+    <div>
+      {loading ? (
+        <div>Loading...</div>
+      ) : error ? (
+        <div>error: {error}</div>
+      ) : (
+        <div>
+          <label htmlFor="cluster_replica">Cluster Replica </label>
+          <select
+            id="cluster_replica"
+            name="cluster_replica"
+            onChange={(event) => {
+              const clusterReplica = event.target.value;
+              const clusterReplicaSplit = clusterReplica.split('|');
+              setCurrentClusterName(clusterReplicaSplit[0]);
+              setCurrentReplicaName(clusterReplicaSplit[1]);
+            }}
+            defaultValue={`${currentClusterName}|${currentReplicaName}`}>
+            {sqlResponse.map((v) => (
+              <option key={`${v[0]}|${v[1]}`} value={`${v[0]}|${v[1]}`}>
+                {`${v[0]}.${v[1]}`}
+              </option>
+            ))}
+          </select>
+          <Views clusterName={currentClusterName} replicaName={currentReplicaName} />
+        </div>
+      )}
+    </div>
+  );
 }
 
-function Views() {
+function Views(props) {
+  const [currentDataflow, setCurrentDataflow] = useState(null);
+  const [includeSystemCatalog, setIncludeSystemCatalog] = useState(false);
+  const [records, setRecords] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
   useEffect(() => {
     const search = new URLSearchParams(location.search);
     const dataflow = search.get('dataflow');
     if (dataflow) {
-      setCurrent(dataflow);
+      setCurrentDataflow(dataflow);
     }
     setIncludeSystemCatalog(search.get('system_catalog') === 'true');
   }, []);
 
-  const [current, setCurrent] = useState(null);
-  const [includeSystemCatalog, setIncludeSystemCatalog] = useState(false);
+  useEffect(() => {
+    setCurrentDataflow(null);
+  }, [props]);
 
-  const where_fragment = includeSystemCatalog ? `` : `WHERE name NOT LIKE 'Dataflow: mz_catalog.%'`;
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    if (currentDataflow) {
+      params.set('dataflow', currentDataflow);
+    } else {
+      params.delete('dataflow');
+    }
+    window.history.replaceState({}, '', `${location.pathname}?${params}`);
```

Comment:
I'm 100% not adding any of the URL goo if it becomes tedious. I've always had weirdo edge cases when I try to do this, and these memory pages don't need pretty URLs.

Reply:
Yeah it wasn't too bad here but also happy to rip it out if it's not worth it